File: Golang/fibonacci-number.go
// Time:
// Space:
package main

func fib(n int) int {
	if n < 2 {
		return n
	}

	pd := make([]int, n+1)
	pd[0], pd[1] = 0, 1

	for i := 2; i < n+1; i++ {
		pd[i] = pd[i-1] + pd[i-2]
	}

	return pd[n]
}

File: Python/fibonacci-number.py
#Time:
#Space:

class Solution:
    def fib(self, n: int) -> int:
        predefined = [0, 1]

        if n < 2:
            return n

        for i in range(2, n+1):
            s = predefined[i-1] + predefined[i-2]
            predefined.append(s)

        return predefined[n]